go/vt/vtadmin/cluster/cluster.go

```diff
@@ -200,6 +204,136 @@ func (c *Cluster) parseTablet(rows *sql.Rows) (*vtadminpb.Tablet, error) {
 	return tablet, nil
 }
 
+// FindWorkflowsOptions is the set of options for FindWorkflows requests.
+type FindWorkflowsOptions struct {
+	ActiveOnly      bool
+	IgnoreKeyspaces sets.String
+	Filter          func(workflow *vtadminpb.Workflow) bool
+}
+
+// FindWorkflows returns a list of Workflows in this cluster, across the given
+// keyspaces and filtering according to the options passed in.
+//
+// If the list of keyspaces to check is empty, then FindWorkflows will use the
+// result of GetKeyspaces to search all keyspaces in the cluster. In this case,
+// opts.IgnoreKeyspaces is respected.
+//
+// Callers should use this function when they want more fine-grained filtering,
+// and GetWorkflows when they just want to filter on keyspace name.
+func (c *Cluster) FindWorkflows(ctx context.Context, keyspaces []string, opts FindWorkflowsOptions) ([]*vtadminpb.Workflow, error) {
+	span, ctx := trace.NewSpan(ctx, "Cluster.FindWorkflows")
+	defer span.Finish()
+
+	AnnotateSpan(c, span)
+	span.Annotate("active_only", opts.ActiveOnly)
+
+	return c.findWorkflows(ctx, keyspaces, opts)
+}
+
+func (c *Cluster) findWorkflows(ctx context.Context, keyspaces []string, opts FindWorkflowsOptions) ([]*vtadminpb.Workflow, error) {
```

Comment:
Note to reviewers: this function is the bulk of the logic; all (non-error) codepaths end up calling this function.